server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -313,26 +313,18 @@ public void testDispatchDoesNotRequireContentTypeForRequestsWithoutContent() {
 
     public void testDispatchFailsWithPlainText() {
         String content = randomAlphaOfLengthBetween(1, BREAKER_LIMIT.bytesAsInt());
-        FakeRestRequest fakeRestRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY)
+        FakeRestRequest request = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY)
             .withContent(new BytesArray(content), null).withPath("/foo")
-            .withHeaders(Collections.singletonMap("Content-Type", Collections.singletonList("text/plain"))).build();
-        AssertingChannel channel = new AssertingChannel(fakeRestRequest, true, RestStatus.NOT_ACCEPTABLE);
-        restController.registerHandler(RestRequest.Method.GET, "/foo", new RestHandler() {
-            @Override
-            public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
-                channel.sendResponse(new BytesRestResponse(RestStatus.OK, BytesRestResponse.TEXT_CONTENT_TYPE, BytesArray.EMPTY));
-            }
-        });
-
-        assertFalse(channel.getSendResponseCalled());
-        restController.dispatchRequest(fakeRestRequest, channel, new ThreadContext(Settings.EMPTY));
-        assertTrue(channel.getSendResponseCalled());
+            .withHeaders(contentTypeHeader("text/plain"))
+            .build();
+        registerHandlerThenMakeRequestAndCheckHandled(request,  RestStatus.NOT_ACCEPTABLE, "/foo");
```

Comment:
lots of copy /pasted code, extracted all the boilerplate and fixed existing and reused for my own new tests.